examples/api/lib/painting/image_provider/image_provider.0.dart

```diff
@@ -0,0 +1,113 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+import 'dart:ui' as ui;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+
+@immutable
+class CustomNetworkImage extends ImageProvider<Uri> {
+  const CustomNetworkImage(this.url);
+
+  final String url;
+
+  @override
+  Future<Uri> obtainKey(ImageConfiguration configuration) {
+    final Uri result = Uri.parse(url).replace(
+      queryParameters: <String, String>{
+        'dpr': '${configuration.devicePixelRatio}',
+        'locale': '${configuration.locale?.toLanguageTag()}',
+        if (configuration.platform == null)
+          'platform': null.toString(),
+        if (configuration.platform != null)
+          'platform': configuration.platform!.toString().substring('$TargetPlatform'.length + 1),
+        'width': '${configuration.size?.width}',
+        'height': '${configuration.size?.height}',
+        if (configuration.textDirection == null)
+          'bidi': null.toString(),
+        if (configuration.textDirection == TextDirection.ltr)
+          'bidi': 'ltr',
+        if (configuration.textDirection == TextDirection.rtl)
+          'bidi': 'rtl',
+      },
+    );
+    return SynchronousFuture<Uri>(result);
+  }
+
+  static HttpClient get _httpClient {
+    HttpClient? client;
+    assert(() {
+      if (debugNetworkImageHttpClientProvider != null) {
+        client = debugNetworkImageHttpClientProvider!();
+      }
+      return true;
+    }());
+    return client ?? HttpClient()..autoUncompress = false;
+  }
+
+  @override
+  ImageStreamCompleter loadImage(Uri key, ImageDecoderCallback decode) {
+    final StreamController<ImageChunkEvent> chunkEvents = StreamController<ImageChunkEvent>();
+    debugPrint('Fetching "$key"...');
+    return MultiFrameImageStreamCompleter(
+      codec:
+        _httpClient.getUrl(key)
```

Comment:
nit: formatting here seems off. Maybe move this to the previous line (or indent the `.then` lines further)?

Reply:
done